pkg/plugin/plugin.go

```diff
@@ -0,0 +1,42 @@
+// package plugin provides the plugin interface for generating SPDX SBOMs
+package plugin
+
+import (
+	"errors"
+
+	"github.com/spdx/tools-golang/spdx/common"
+)
+
+// Plugin is the interface a plugin must implement
+// GetSpdxDocument returns an object of type AnyDocument
+type Plugin interface {
+	GetSpdxDocument() (common.AnyDocument, error)
+}
+
+// plugins stores a mapping of the plugin name and its
+// corresponding Plugin object
+var plugins = make(map[string]Plugin)
```

Comment:
How is the `name` we're using to key supposed to be used? Will we support more than one plugin for each ecosystem? I see it is the argument to GetPlugin() but should we have a keying scheme tied to the specific ecosystem?

I expect to look for plugins based on what the generator finds in the code repository, so more than a name should this be an ecocystem label or another moniker?